### ISSUE TITLE
Errors in feature_extraction/main.py

### ISSUE THREAD

**Issue by ntorqulu** [ntorqulu/taxo_classification#17]
Things to discuss:
1. in kmer_numbering_for_one_kmer():

self.multiplyBy = 4**np.arrange(k-1, -1, -1) # numpy array
digits = torch.tensor(digits) # tensor
numbering = (digits * self.multiplyBy).sum() # ERROR! multiplying numpy with tensor

FIX:
- In kmerize_one_seq, define:
multiplyBy = 4 ** np.arange(k-1, -1, -1) # do not treat it as instance attribute
this_numbering = self.kmer_numbering_for_one_kmer(this_kmer, multiplyBy) # pass it in the function call

2. class initialization def __init__(self):
- include:
self.sequences = None
- delete:
return None

3. Inconsistent return types:
- coding_kmer -> tensor
- coding_one_hot_4rowMatrix -> numpy array
- coding_one_hot_bit -> tensor
Should be all standarized to tensors

4. In coding_one_hot_4rowMatrix, convert list to tensor for consistency
- Question: This would require padding for variable-length sequences?
